### PR TITLE
[CR-1895] Separate success and close callbacks

### PR DESCRIPTION
**Pull request by Arlanthir**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,7 +61,6 @@ checkoutInstance.unmount()
 
 ```javascript
 function mySuccessHandler(successInfo) {
-  checkoutInstance.unmount()
   /** Show your own thank you message and/or do something with the payment info. */
 }
```
Comment (antonioliv) [inline README.md:65]: I think there is some changes to be done in the README, like for example the handler onClose

File: src/index.spec.ts [hunk 2]
```diff
@@ -311,6 +294,25 @@ describe('SDK', () => {
     checkout.unmount()
   })
 
+  test('calls onClose', () => {
```
Comment (antonioliv) [inline src/index.spec.ts:297]: I think we should add a test for each display option, for inline and for popup.

File: src/index.spec.ts [hunk 3]
```diff
@@ -337,8 +339,26 @@ describe('SDK', () => {
...
+          id: '2fdc12ca-d600-4ef4-be51-be1626cc1329',
+          type: 'single',
+          payment: {
+            id: '4e5a2766-e010-4ed0-8bc0-eea57fb30d63',
+            method: 'cc',
+            status: 'authorised',
+            value: 120,
+          },
+        }
+      },
```
[the remark this answers is not in the document]
Comment (Arlanthir) [reply src/index.spec.ts:361]: To verify that it's removing the event handler, we send another message and confirm it didn't save it.

[the remark this answers is not in the document]
Comment (antonioliv) [reply src/index.spec.ts:361]: I see, the id of the "checkoutResult" of the `secondMessage` differs from the `checkoutResult` initialized at the top of the spec.

**Review by Espina2 — APPROVED** [review]
LGTM

**Review by antonioliv — APPROVED** [review]
LGTM!

**Pull request merged**